test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -210,27 +210,27 @@ Describe "Basic FileSystem Provider Tests" -Tags "CI" {
                 $protectedPath2 = Join-Path $protectedPath "Install"
                 $newItemPath = Join-Path $protectedPath "foo"
             }
-            $errFile = "$testdrive\error.txt"
-            $doneFile = "$testdrive\done.txt"
-        }
-        AfterEach {
-            Remove-Item -Force $errFile -ErrorAction SilentlyContinue
-            Remove-Item -Force $doneFile -ErrorAction SilentlyContinue
         }
 
         It "Access-denied test for '<cmdline>" -Skip:(-not $IsWindows) -TestCases @(
-            @{cmdline = "Get-Item $protectedPath2"; expectedError = "ItemExistsUnauthorizedAccessError,Microsoft.PowerShell.Commands.GetItemCommand"}
-            @{cmdline = "Get-ChildItem $protectedPath"; expectedError = "DirUnauthorizedAccessError,Microsoft.PowerShell.Commands.GetChildItemCommand"}
-            @{cmdline = "New-Item -Type File -Path $newItemPath"; expectedError = "NewItemUnauthorizedAccessError,Microsoft.PowerShell.Commands.NewItemCommand"}
-            @{cmdline = "Rename-Item -Path $protectedPath -NewName bar"; expectedError = "RenameItemIOError,Microsoft.PowerShell.Commands.RenameItemCommand"},
-            @{cmdline = "Move-Item -Path $protectedPath -Destination bar"; expectedError = "MoveDirectoryItemIOError,Microsoft.PowerShell.Commands.MoveItemCommand"},
-            @{cmdline = "Remove-Item -Path $protectedPath"; expectedError = "RemoveItemUnauthorizedAccessError,Microsoft.PowerShell.Commands.RemoveItemCommand"}
+            @{cmdline = "Get-Item $protectedPath2"; expectedError = "ItemExistsUnauthorizedAccessError,Microsoft.PowerShell.Commands.GetItemCommand"; testId = 'GetItem'}
+            @{cmdline = "Get-ChildItem $protectedPath"; expectedError = "DirUnauthorizedAccessError,Microsoft.PowerShell.Commands.GetChildItemCommand"; testId = 'GetChildItem'}
+            @{cmdline = "New-Item -Type File -Path $newItemPath"; expectedError = "NewItemUnauthorizedAccessError,Microsoft.PowerShell.Commands.NewItemCommand"; testId = 'NewItem'}
+            @{cmdline = "Rename-Item -Path $protectedPath -NewName bar"; expectedError = "RenameItemIOError,Microsoft.PowerShell.Commands.RenameItemCommand"; testId = 'RenameItem'},
+            @{cmdline = "Move-Item -Path $protectedPath -Destination bar"; expectedError = "MoveDirectoryItemIOError,Microsoft.PowerShell.Commands.MoveItemCommand";  testId = 'MoveItem'},
+            @{cmdline = "Remove-Item -Path $protectedPath"; expectedError = "RemoveItemUnauthorizedAccessError,Microsoft.PowerShell.Commands.RemoveItemCommand"; testId = 'RemoveItem'}
         ) {
-            param ($cmdline, $expectedError)
+            param ($cmdline, $expectedError, $testId)
```

Comment:
Rather than adding a TestId param, you can either just use `cmdline` or use $cmdline.split(' ')[0] to get the same as what you have.  Using the cmdline may be better if new test cases for the same command get added in the future.

Reply:
Also, line 238 to check existence of $errfile should be removed since it'll always exist since you're creating it before the test.

Reply:
I considered using split but it won't solve the problem; splitting on space gives the cmdlet name. I also considered using the expected error id but that could end up colliding as well.  The test id was the most straight forward solution. I'll add a comment to the test cases to call the unique requirement out specifically.

As to the errFile test, I'll remove it.

Reply:
@SteveL-MSFT Does that work for you?

Reply:
I think it would be simpler to just use the entirety of $cmdline in place of $testid (just put it in quotes on line 229).

Reply:
That's not reliable; there could be characters in the command-line that are not valid for a file name. For example, the '|' is not a valid filename character.

Reply:
The testid parameter is used to generate the test-specific error and done file names. This is being done to address the issue of the launched PowerShell process taking longer than the timeout to shutdown. As observed in the nightly build, this can cause a race between the current test and next test if the PowerShell process doesn't exit before the next test starts.

I've changed the parameter name to fileNameBase for clarity.